segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ContainerMetadataUpdateTransaction.java

```diff
@@ -318,31 +317,29 @@ void preProcessOperation(Operation operation) throws ContainerException, StreamS
             }
         }
 
-        if (operation instanceof StorageOperation) {
-            if (operation instanceof StreamSegmentAppendOperation) {
-                segmentMetadata.preProcessOperation((StreamSegmentAppendOperation) operation);
-            } else if (operation instanceof StreamSegmentSealOperation) {
-                segmentMetadata.preProcessOperation((StreamSegmentSealOperation) operation);
-            } else if (operation instanceof MergeTransactionOperation) {
-                MergeTransactionOperation mbe = (MergeTransactionOperation) operation;
-                SegmentMetadataUpdateTransaction transactionMetadata = getSegmentUpdateTransaction(mbe.getTransactionSegmentId());
-                transactionMetadata.preProcessAsTransactionSegment(mbe);
-                segmentMetadata.preProcessAsParentSegment(mbe, transactionMetadata);
-            }
-        } else if (operation instanceof MetadataOperation) {
-            if (operation instanceof StreamSegmentMapOperation) {
-                preProcessMetadataOperation((StreamSegmentMapOperation) operation);
-            } else if (operation instanceof TransactionMapOperation) {
-                preProcessMetadataOperation((TransactionMapOperation) operation);
-            } else if (operation instanceof UpdateAttributesOperation) {
-                segmentMetadata.preProcessOperation((UpdateAttributesOperation) operation);
-            } else if (operation instanceof MetadataCheckpointOperation) {
-                // MetadataCheckpointOperations do not require preProcess and accept; they can be handled in a single stage.
-                processMetadataOperation((MetadataCheckpointOperation) operation);
-            } else if (operation instanceof StorageMetadataCheckpointOperation) {
-                // StorageMetadataCheckpointOperation do not require preProcess and accept; they can be handled in a single stage.
-                processMetadataOperation((StorageMetadataCheckpointOperation) operation);
-            }
+        if (operation instanceof StreamSegmentAppendOperation) {
```

Comment:
For my understanding only: why are we not making a distinction between `StorageOperation` and `MetadataOperation` any longer?

Reply:
it was redundant. It was checking the type of the operation twice, and the first one was redundant. It wasn't doing anything special, so I simplified the code